services/cards-consultation/src/main/java/org/opfab/cards/consultation/services/ConnectedEntityPreviewService.java

```diff
@@ -0,0 +1,51 @@
+/* Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * See AUTHORS.txt
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ * This file is part of the OperatorFabric project.
+ */
+
+package org.opfab.cards.consultation.services;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import org.opfab.cards.consultation.model.LightCard;
+import org.opfab.users.model.CurrentUserWithPerimeters;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ConnectedEntityPreviewService {
+
+    private final CardSubscriptionService cardSubscriptionService;
+
+    public ConnectedEntityPreviewService(CardSubscriptionService cardSubscriptionService) {
+        this.cardSubscriptionService = cardSubscriptionService;
+    }
+
+    public List<String> getConnectedRecipients(LightCard lightcard) {
+        List<String> connectedRecipients = new ArrayList<>();
+        
+        List<String> cardRecipients = lightcard.getEntityRecipients(); 
+        List<String> cardRecipientsForInformation = lightcard.getEntityRecipientsForInformation();
+
+        List<String> cardTotalRecipients = new ArrayList<>(cardRecipients);
+        cardTotalRecipients.addAll(cardRecipientsForInformation);
+        Collection<CardSubscription> connections = this.cardSubscriptionService.getSubscriptions();
+        for (CardSubscription cardSubscription : connections) {
+            CurrentUserWithPerimeters userWithPerimeters = cardSubscription.getCurrentUserWithPerimeters();
+            if (CardRoutingUtilities.checkIfUserMustBeNotifiedForThisProcessState(lightcard.getProcess(), lightcard.getState(), userWithPerimeters)) {
```

Comment:
The method checkIfUserMustBeNotifiedForThisProcessState does not test all the conditions for the user to be notify

You have to use checkIfUserMustReceiveTheCard()

But he method take as params a JSON Object so you need to "cut" this method in two to obtain a method with params : groupRecipients,entityRecipients,userRecipients & currentUserWithPerimeters